simtools/configuration/commandline_parser.py

```diff
@@ -237,7 +242,10 @@ def initialize_telescope_model_arguments(self, add_model_version=True, add_teles
             Set to allow a telescope name argument.
         """
 
-        _job_group = self.add_argument_group("telescope model")
+        if add_telescope:
+            _job_group = self.add_argument_group("telescope model")
+        else:
+            _job_group = self.add_argument_group("site model")
```

Comment:
if add_telescope is hard wired in l.61 we don't need this check here. I suggest either propagating the option to pass add_telescope to the application or get rid of the if statement.

Reply:
After looking at the remaining changes, getting rid of this if statement sounds like a better option to me.

Reply:
But we still have line 59:

- with `add_telescope=True` we allow to specify `--telescope` and `--site`  on the command line
- with `add_telescope=False`, we allow only `site` (this is important for applications which depend on the site only

Or did I misunderstand your comment?

Reply:
Ah, that is all good (I looked at a different function).